### PR TITLE
Add mmcls2torchserve

### PR DESCRIPTION
**Pull request by daavoo**
This P.R. enables serving MMClassification models with TorchServe. A new section Model Serving has been added to tutorials.

### CODE REVIEW

**Comment by mzr1996** [comment]
Hello, thanks for your contributing!
It's a good idea to export mmcls model to torchserve, but I think the docker deployment is a little heavy for mmcls repo, and how to deploy a torchserve model to docker is not much related to mmcls repo.
Since there is a complete deployment tutorial in [serve repo](https://github.com/pytorch/serve/blob/master/docker/README.md), I think just linking it in the tutorial is enough, including docker deployment and local deployment.
What do you think? @ycxioooong

**Comment by daavoo** [comment]
> Hello, thanks for your contributing!
> It's a good idea to export mmcls model to torchserve, but I think the docker deployment is a little heavy for mmcls repo, and how to deploy a torchserve model to docker is not much related to mmcls repo.
> Since there is a complete deployment tutorial in [serve repo](https://github.com/pytorch/serve/blob/master/docker/README.md), I think just linking it in the tutorial is enough, including docker deployment and local deployment.
> What do you think? @ycxioooong

I agree with your points, the only thing might be that, as far as I know, if someone follows the official serve instructions (i.e. docker deployment) the mmcls models won't work because mmcv/mmcls won't be installed in the environment.

**Comment by mzr1996** [comment]
> > Hello, thanks for your contributing!
> > It's a good idea to export mmcls model to torchserve, but I think the docker deployment is a little heavy for mmcls repo, and how to deploy a torchserve model to docker is not much related to mmcls repo.
> > Since there is a complete deployment tutorial in [serve repo](https://github.com/pytorch/serve/blob/master/docker/README.md), I think just linking it in the tutorial is enough, including docker deployment and local deployment.
> > What do you think? @ycxioooong
> 
> I agree with your points, the only thing might be that, as far as I know, if someone follows the official serve instructions (i.e. docker deployment) the mmcls models won't work because mmcv/mmcls won't be installed in the environment.

Well, maybe I need some time to be familiar with torchserve to evaluate and improve this function and tutorial better. Thanks for your contribution again.

File: tools/deployment/mmcls2torchserve.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        args = Namespace(
+            **{
+                'model_file': f'{tmpdir}/config.py',
+                'serialized_file': checkpoint_file,
+                'handler': f'{Path(__file__).parent}/mmcls_handler.py',
+                'model_name': model_name or Path(checkpoint_file).stem,
+                'version': model_version,
+                'export_path': output_folder,
+                'force': force,
+                'requirements_file': None,
```
Comment (mzr1996) [inline tools/deployment/mmcls2torchserve.py:61]: Here, torchserve provides an option [`requirements_file`](https://github.com/pytorch/serve/tree/master/model-archiver#model-specific-custom-python-requirements) to auto-install requirements in the deployment environment, may be we can use this?

Comment (ycxioooong) [reply tools/deployment/mmcls2torchserve.py:61]: for mmcv-full, installation with requirements_file may cause errors.

**Review by ycxioooong — APPROVED** [review]
LGTM

**Pull request merged**